pkg/cmd/server/origin_master.go

```diff
@@ -83,10 +85,18 @@ func (cfg Config) BuildOriginMasterConfig() (*origin.MasterConfig, error) {
 		return nil, err
 	}
 
+	dnsAddr := cfg.DNSBindAddr
```

Comment:
This validation should happen in a separate method call for GetDNSAddr like we do for other addresses that require detection.  https://github.com/openshift/origin/blob/master/pkg/cmd/server/config.go#L123 as an example.